src/game/objects/common.c

```diff
@@ -5,6 +5,9 @@
 #include "game/inventory.h"
 #include "game/items.h"
 #include "game/lara.h"
+#include "game/objects/general/keyhole.h"
+#include "game/objects/general/puzzle_hole.h"
+#include "game/objects/traps/midas_touch.h"
```

Comment:
We shouldn't include these inside `common.c` – generalizations shouldn't depend on specific implementations.

To deal with `Object_CheckForHole` having to know too much about specific items, I propose another series of refactors (that should be landed in a separate PR first, I can do that):

- convert `int16_t[12]` to a proper `OBJECT_BOUNDS`  struct
- introduce `OBJECT_BOUNDS *Object_GetBounds(GAME_OBJECT_ID object_num)`
- introduce `OBJECT_BOUNDS *(*get_bounds)(void);` in `OBJECT_INFO`
- introduce specializations in each object type that needs it
- now only call `Object_GetBounds(object_num)` without having to know the exact implementation

Reply:
I was thinking of a `bool (*is_interactable)(int16_t item_num)` in `OBJECT_INFO` and that way you could leave the specifics to the keyhole, puzzle slot etc, but to be honest I prefer @rr-'s solution instead as it could be used over and above the current case.